src/NServiceBus.Core/Persistence/InMemory/InMemoryTransactionalSynchronizedStorageAdapter.cs

```diff
@@ -22,13 +22,16 @@ public bool TryAdapt(OutboxTransaction transaction, out CompletableSynchronizedS
 
         public bool TryAdapt(TransportTransaction transportTransaction, out CompletableSynchronizedStorageSession session)
         {
-            var ambientTransaction = transportTransaction as AmbientTransaction;
-            if (ambientTransaction != null)
+            if (transportTransaction.Data.ContainsKey("AmbientTransaction"))
             {
-                var transaction = new InMemoryTransaction();
-                session = new InMemorySynchronizedStorageSession(transaction);
-                ambientTransaction.Transaction.EnlistVolatile(new EnlistmentNotification(transaction), EnlistmentOptions.None);
-                return true;
+                var ambientTransaction = transportTransaction.Data["AmbientTransaction"] as Transaction;
```

Comment:
can we make keys more explicit like 'Msmq.AmbitentTransaction' to minimize collisions?